worksheet_B/Mandelbrot.cpp

```diff
@@ -12,44 +90,133 @@ int main()
 	CImgDisplay display(image, "Mandelbrot Set");
 
 	// Minimum and maximum coordinates for the fractal
-	const double minX = -2, maxX = 1, minY = -1.5, maxY = 1.5;
+			double minX = -2, maxX = 1, minY = -1.5, maxY = 1.5;
+
+	// Zooming variables
+			bool zoomIn;
+			bool recalculateBounds = false;
+			double zoomMagnitude = 1.5;
+			double currentSideLength = maxX - minX;
+			const double originalSideLength = currentSideLength;
 
-	// Generate the image
-	for (int pixelY = 0; pixelY < image.height(); pixelY++)
+while (true)
 	{
-		// TODO: Map the y coordinate into the range minY to maxY
-		//double y0 =
 
-		for (int pixelX = 0; pixelX < image.width(); pixelX++)
+		// Requires holding the button down, odd event handling
+		if (GetKeyState(VK_LBUTTON) < 0)
+		{
+			zoomIn = true;
+			recalculateBounds = true;
+		}
+		else if (GetKeyState(VK_RBUTTON) < 0)
+		{
+			zoomIn = false;
+			recalculateBounds = true;
+		}
+
+		if (recalculateBounds)
+		{
+			recalculateBounds = false;
+
+			// Get mouse position
+			POINT mousePosition;
+			GetCursorPos(&mousePosition);
+
+			// Get local in-window coordinates of mouse cursor for zoom target
+			double localPositionX = mousePosition.x - display._window_x;
+			double localPositionY = mousePosition.y - display._window_y;
+
+			// Compare against window dimensions to determine mouse cursor position in terms of data bounds
+			double ratioX = localPositionX / display._window_width;
+			double dataRangeCursorPositionX = (ratioX * currentSideLength) + minX;
+
+			double ratioY = localPositionY / display._window_height;
+			double dataRangeCursorPositionY = (ratioY * currentSideLength) + minY;
+
+			// Apply zoom modifier to alter currentSideLength
+			if (zoomIn)
+				currentSideLength /= zoomMagnitude;
+			else
+				currentSideLength *= zoomMagnitude;
+
+			// Determine new minX, minY values for zoomed section with mouse cursor as new centre
+			minX = dataRangeCursorPositionX - (currentSideLength / 2);
+			minY = dataRangeCursorPositionY - (currentSideLength / 2);
+
+			// Determine new maxX, maxY values
+			maxX = minX + currentSideLength;
+			maxY = minY + currentSideLength;
+
+			// Display zoom level to console
+			double zoomLevel = originalSideLength / currentSideLength;
+			std::cout << "Zoom magnitude: " << zoomLevel << std::endl;
+		}
+
+			// Generate the image
+		for (double pixelY = 0; pixelY < image.height(); pixelY++)
```

Comment:
This works, but having a `double` as a loop counter is non-standard. Better to keep it as an `int` and cast to `double` when necessary